src/charm.py

```diff
@@ -242,11 +243,15 @@ def grpc_server_url(self):
         """
         if external_host := self.ingress.grpc_external_host:
             # this does not include any scheme.
-            return f"{external_host}:{Nginx.parca_http_server_port}"
+            return f"{external_host}:{Nginx.parca_grpc_server_port}"
         return f"{self._fqdn}:{Nginx.parca_grpc_server_port}"
 
     @property
-    def _scheme(self) -> str:
+    def _scheme(self):
```

Comment:
how about:
```suggestion
    def _ingress_scheme(self):
```
to make the distinction clear?

Reply:
its not really ingress scheme, since if ingress is not available, we'll use the internal one

Reply:
I changed the other one to `internal_scheme` since it only returns the internal one

Reply:
Added a docstring